### ISSUE TITLE
further defining user preferences about email submission

### ISSUE THREAD

**Issue by volodymyrss** [oda-hub/dispatcher-app#184]
it is possible to prevent email submission in some cases with user preferences in token

but it appears to be useful to additionally control them per-request.
maybe there could be an additional parameter passed. the problem is that it would have to be passed around also in dispatcher. So I am thinking we could pass it only once in run_analysis, and then internally create new token without no-email preference.

also, it's not clear how to get the no-email token in frontend.

**Comment by volodymyrss** [comment]
it needs to be done for internal use, it's not critical at this time otherwise, but still could be useful.

**Comment by burnout87** [comment]
Perhaps it could be something the user can choose directly from the front-end. 

For example, we could have a couple of checkboxes:
- receive email updates
- receive email when the job is completed in case of a long request

Those would be initialised depending on how the token is set for the user.

**Comment by volodymyrss** [comment]
> Perhaps it could be something the user can choose directly from the front-end.

yes, that was the last comment of the issue: frontend should be able to define it. this issues should be made in frontend-astrooda and dealt-with as needed.

but, in addition to that, api-only interactions may only be customly skip email sometimes.

**Comment by burnout87** [comment]
> but, in addition to that, api-only interactions may only be customly skip email sometimes.

So, also in this case, an additional set of email-related options should be available

**Comment by volodymyrss** [comment]
> > but, in addition to that, api-only interactions may only be customly skip email sometimes.
> 
> So, also in this case, an additional set of email-related options should be available

We could use the same options already defined in the token.

**Comment by volodymyrss** [comment]
The internal part of this already works very well for me. Other users would benefit.
